app/jobs/hyku_addons/reindex_model_job.rb

```diff
@@ -35,14 +35,19 @@ def perform(klass, cname, limit: 25, page: 1, options: {})
     private
 
       def mint_doi(work)
-        return if work.doi.present?
+        return if work.doi.present? || work.visibility != "open" || workflow_state(work)&.workflow_state_name != "deposited"
```

Comment:
I find that when the guard clause starts to have more than one or two clauses to it, then it is better to add it to its own method. With this example there are positive and negative checks too. I would use something like this: 

```
return unless can_mint_for?(work)

def can_mint_for?(work)
    work.doi.blank? || work.visibility == "open" || workflow_state(work)&.workflow_state_name == "deposited"
end
```

Reply:
Thanks will update it

Reply:
I made the change to use unless with these conditions:

    work.doi.blank? || work.visibility == "open" || workflow_state(work)&.workflow_state_name == "deposited"
    
but the test failed so I reverted it back to what it was  but extracted it into a method like you did.
I thought i should let you know